### ISSUE TITLE
Can't Group By Content Type with Collapse='TRUE'

### ISSUE THREAD

**Issue by sympmarc** [SharePoint/sp-dev-docs#5188]
#### Category
- [ ] Question
- [ ] Typo
- [X] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

We'd like to be able to set up view which are grouped by Content Type and are collapsed by default. So we use some JSON like this:

``` json
        {
          "verb": "addSPView",
          "name": "New View",
          "viewFields": [
            "ContentType",
            "DocIcon",
            "LinkFilename",
            "ReportDate",
            "Modified",
            "Editor"
          ],
          "query": "<OrderBy><FieldRef Name=\"FileLeafRef\" /></OrderBy><GroupBy Collapse='TRUE'><FieldRef Name=\"ContentType\" Ascending=\"TRUE\" /></GroupBy>",
          "rowLimit": 30,
          "isPaged": true,
          "makeDefault": true
        }
```

#### Observed Behavior

The resulting view when we apply the Site Design which reference the Site Script snippet from above is grouped by Content Type, but it displays the ContentTypeId rather than the Content Type name.

![image](https://user-images.githubusercontent.com/1295627/72831580-280a4280-3c51-11ea-9392-ad762778077c.png)

Using the SPInsider Chrome extension, I can see the resulting ViewXML is correct:

``` xml
<OrderBy><FieldRef Name=\"FileLeafRef\" /></OrderBy><GroupBy Collapse='TRUE'><FieldRef Name=\"ContentType\" Ascending=\"TRUE\" /></GroupBy>
```

If we *don't* add `Collapse='TRUE'`, the Content Type name is shown correctly.

![image](https://user-images.githubusercontent.com/1295627/72833518-fabf9380-3c54-11ea-8c44-c264c72fedef.png)

#### Steps to Reproduce

Use the JSON snippet above to add a new view to a list or library with or without the  `Collapse='TRUE'` setting.

**Comment by rneel** [comment]
@sympmarc Were you able to ever find a solution or work-around to this?  We have successfully grouped a list by Content Type, but now we can't default it to collapsed.

**Comment by sympmarc** [comment]
No, I never heard anything back. If I remember correctly, we either abandoned the approach or came up with a workaround (maybe running PowerShell). It's sad how none of the Site Designs issues here ever get any attention.

**Comment by Heinrich84HH** [comment]
hello @sympmarc i have a solution for this problem, i use pnp powershell to export/import views for a libary. in default i have the same problem with the IDs instead of the names. But i have added in the view xml a aditional "GrouBy" and put the "Collapse ="True" inthere: 
![grafik](https://github.com/SharePoint/sp-dev-docs/assets/153530275/7932abe2-7f77-471e-a1c8-964eb491ebfd)
In my case, i have a view with two groups, first by the column "Prozess" and second for the Contenttype.

**Comment by ScottPRagen** [comment]
We also get the Content Type ID in Hexadecimal, but we get one occurrence of each group when trying to group and collapse by Content Type using the following query:
`<GroupBy GroupLimit='30' Collapse='TRUE'><FieldRef Name='ContentType'/></GroupBy>`
![Image](https://github.com/user-attachments/assets/c8b98f4e-ea40-4b87-ab9e-29d2ce199ca6)

However if I use the suggestion by @Heinrich84HH it shows the names of the Content Types, but there are multiple occurrences and when expanding one occurrence, it expands the others by using the following query:
`<GroupBy GroupLimit='30'><FieldRef Name='ContentType'/></GroupBy><GroupBy Collapse='TRUE'></GroupBy>`
![Image](https://github.com/user-attachments/assets/e78417c9-6ab2-43ce-af31-0f89023a61d3)

It also does the same if I use ID="text", but I've not used CAML before, so not sure why or how to fix it.

Hope this can be looked at, our client is keen to use it.

Thanks,

Scott
